mewarpx/mewarpx/assemblies.py

```diff
@@ -294,3 +299,184 @@ def _install_in_simulation(self):
+            install_in_simulation (bool): If True and the Assembly is an
+                embedded boundary it will be included in the WarpX simulation
+        """
+        super(Rectangle, self).__init__(V=V, T=T, WF=WF, name=name)
+        self.center_x = float(center_x)
+        self.center_z = float(center_z)
+        self.length_x = float(length_x)
+        self.length_z = float(length_z)
+        self.xmin = float(center_x - length_x/2)
+        self.xmax = float(center_x + length_x/2)
+        self.zmin = float(center_z - length_z/2)
+        self.zmax = float(center_z + length_z/2)
+
+        self.scaled_h = (
+            2.0 * max(self.length_x, self.length_z) / min(self.length_x, self.length_z)
+        )
+
+        if self.length_x <= self.length_z:
+            self.region_normals = np.array([
+                (0, -1), (1, 0), (0, 1), (-1, 0)
+            ])
+        else:
+            self.region_normals = np.array([
+                (-1, 0), (0, -1), (1, 0), (0, 1)
+            ])
+
+        self.transformation_matrix = self._get_transformation_matrix()
+        # negative means outside of object
+
+        self.implicit_function = (
+            f"-max(max(x-({self.xmax}),({self.xmin})-x),max(z-({self.zmax}),({self.zmin})-z))"
+        )
+        self.scraper_label = 'eb'
+
+        if install_in_simulation:
+            self._install_in_simulation()
+
+    def _get_transformation_matrix(self):
+        """Tranform1 is used to move the rectangle to the appropriate location
+        in the plane and scale it to the appropriate size. The rectangle is
+        placed such that the longest side is along the z-axis and the bottom
+        right corner is at (1, -1). This placement makes it simple to pick out
+        positions that are in regions 0 or 1 as well as positions in region 4
+        or 5.
+        """
+        # first transpose the box to the origin
+        transpose1 = np.identity(3)
+        transpose1[0, 2] = (-self.center_x + self.length_x / 2.0)
+        transpose1[1, 2] = (-self.center_z + self.length_z / 2.0)
+
+        # next scale the axes to make zone 0 equilateral with side length 1
+        l = min(self.length_x, self.length_z)
+        scale = np.identity(3)
+        scale[0, 0] = 2.0 / l
+        scale[1, 1] = 2.0 / l
+
+        # next transpose again so the center is at the origin
+        transpose2 = np.identity(3)
+        transpose2[0, 2] = -1
+        transpose2[1, 2] = -1
+
+        # rotate by 90 deg if the length_z is smallest
+        rotate = np.identity(3)
+        if self.length_z < self.length_x:
+            rotate[0,0] = np.cos(np.pi/2)
+            rotate[0,1] = -np.sin(np.pi/2)
+            rotate[1,0] = np.sin(np.pi/2)
+            rotate[1,1] = np.cos(np.pi/2)
+
+        transformation_matrix = np.dot(
+            np.dot(rotate, transpose2),
+            np.dot(scale, transpose1)
+        )
+        return transformation_matrix
+
+    def _transform_coordinates(self, points_in):
+
```

Comment:
```suggestion
```
"No blank line following a def line." is at least Google style guide's approach.